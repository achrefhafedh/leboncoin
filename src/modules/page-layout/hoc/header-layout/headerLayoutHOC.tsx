import React, { SFC, ReactElement, ReactNode } from 'react';
import { curry } from 'ramda';

type HeahderLayoutProps = {
  headerClassName?: string;
  pageContentClassName?: string;
};

export const headerLayoutHOC = curry(
  (header: ReactNode, pageContent: ReactNode) => {
    const HeaderLayout: SFC<HeahderLayoutProps> = ({
      headerClassName,
      pageContentClassName,
    }) =>
      (
        <>
          <div className={headerClassName}>{header}</div>
          <div className={pageContentClassName}>{pageContent}</div>
        </>
      ) as ReactElement<any>;

    return HeaderLayout;
  }
);
