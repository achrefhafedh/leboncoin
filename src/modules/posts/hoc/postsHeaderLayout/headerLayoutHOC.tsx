import React from 'react';

import { headerLayoutHOC } from 'modules/page-layout/hoc/header-layout';
import { Header } from 'modules/page-layout/containers/Header';

const postsHeaderLayoutHOC = headerLayoutHOC(<Header />);

export { postsHeaderLayoutHOC };
