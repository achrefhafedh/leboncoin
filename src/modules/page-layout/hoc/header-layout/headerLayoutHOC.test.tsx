import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { headerLayoutHOC } from './headerLayoutHOC';

configure({ adapter: new Adapter() });

describe('render without crashing', () => {
  let header;
  let pageContent;
  let HeaderLayout: any;

  beforeEach(() => {
    header = <div>Header</div>;
    pageContent = <div>PageContent</div>;
    HeaderLayout = headerLayoutHOC(header, pageContent);
  });

  it('render without crashing', () => {
    const headerHOC = shallow(<HeaderLayout />);
    expect(headerHOC).toMatchSnapshot();
  });
});
