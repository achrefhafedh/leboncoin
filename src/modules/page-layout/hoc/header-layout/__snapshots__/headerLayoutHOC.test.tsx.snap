// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render without crashing render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HeaderLayout />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div>
          <div>
            Header
          </div>
        </div>,
        <div>
          <div>
            PageContent
          </div>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div>
            Header
          </div>,
          "className": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Header",
          },
          "ref": null,
          "rendered": "Header",
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div>
            PageContent
          </div>,
          "className": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "PageContent",
          },
          "ref": null,
          "rendered": "PageContent",
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div>
            <div>
              Header
            </div>
          </div>,
          <div>
            <div>
              PageContent
            </div>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div>
              Header
            </div>,
            "className": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Header",
            },
            "ref": null,
            "rendered": "Header",
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div>
              PageContent
            </div>,
            "className": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "PageContent",
            },
            "ref": null,
            "rendered": "PageContent",
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
